cmd/allocsim gratuitous segfaults

I was ready to blame the recent build changes, but it seems to reproduce even before them.

```
$ go run cmd/allocsim/main.go
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4e02cb6]

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster.(*Node).Kill(0x0)
	/Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster/localcluster.go:434 +0x26
github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster.(*Cluster).Close(0xc4205f83f0)
	/Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster/localcluster.go:139 +0x4c
panic(0x53fa720, 0x632a200)
	/Users/alex/go1.8/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster.(*Node).Start(0xc42061a240)
	/Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster/localcluster.go:410 +0x81f
github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster.(*Cluster).makeNode(0xc4205f83f0, 0x0, 0x7fff5fbffa60, 0x67, 0xc42037c0a0, 0x1, 0x1, 0x0, 0x0, 0x0, ...)
	/Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster/localcluster.go:196 +0x85e
github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster.(*Cluster).Start(0xc4205f83f0, 0x5598500, 0x8, 0x1, 0x7fff5fbffa60, 0x67, 0xc42037c010, 0x1, 0x1, 0x0, ...)
	/Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/cmd/internal/localcluster/localcluster.go:126 +0x36c
main.main()
	/Users/alex/cockroach/pkg/cmd/allocsim/main.go:510 +0xdee
exit status 2
```

It even panics printing the help text...

```
$ go run cmd/allocsim/main.go --help
Usage of /var/folders/0y/zlxqfzj16sgfrxt7p13jqxhw0000gn/T/go-build147236078/command-line-arguments/_obj/exe/main:
  -b int
    	block size (default 1000)
  -duration duration
    	how long to run the simulation for (default 2562047h47m16.854775807s)
  -f string
    	config file that specifies an allocsim workload (overrides -n)
  -log-backtrace-at value
    	when logging hits line file:N, emit a stack trace
  -log-dir value
    	if non-empty, write log files in this directory
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x43e0265]

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/humanizeutil.(*BytesValue).String(0xc42037a350, 0x541ebc0, 0xc42037a350)
	/Users/alex/go/src/github.com/cockroachdb/cockroach/pkg/util/humanizeutil/humanize.go:104 +0x25
flag.isZeroValue(0xc420014cc0, 0xc420013808, 0x7, 0x55c2450)
	/Users/alex/go1.8/src/flag/flag.go:395 +0x102
flag.(*FlagSet).PrintDefaults.func1(0xc420014cc0)
	/Users/alex/go1.8/src/flag/flag.go:469 +0x1b9
flag.(*FlagSet).VisitAll(0xc420072060, 0xc42020f3f0)
	/Users/alex/go1.8/src/flag/flag.go:325 +0x67
flag.(*FlagSet).PrintDefaults(0xc420072060)
	/Users/alex/go1.8/src/flag/flag.go:478 +0x4f
flag.PrintDefaults()
	/Users/alex/go1.8/src/flag/flag.go:501 +0x2d
flag.glob..func1()
	/Users/alex/go1.8/src/flag/flag.go:525 +0xc5
flag.commandLineUsage()
	/Users/alex/go1.8/src/flag/flag.go:963 +0x27
flag.(*FlagSet).usage(0xc420072060)
	/Users/alex/go1.8/src/flag/flag.go:826 +0x2f
flag.(*FlagSet).parseOne(0xc420072060, 0xbb0e570, 0x0, 0xa000005385e80)
	/Users/alex/go1.8/src/flag/flag.go:868 +0x200
flag.(*FlagSet).Parse(0xc420072060, 0xc42000c2b0, 0x1, 0x1, 0xc42020f7b8, 0x4013462)
	/Users/alex/go1.8/src/flag/flag.go:913 +0x60
flag.Parse()
	/Users/alex/go1.8/src/flag/flag.go:941 +0x72
main.main()
	/Users/alex/cockroach/pkg/cmd/allocsim/main.go:408 +0xa1
exit status 2
```

--- a/pkg/cmd/allocsim/main.go
+++ b/pkg/cmd/allocsim/main.go
@@ -255,7 +255,7 @@ func (a *allocSim) rangeInfo() (total int, replicas, leases, leaseTransfers []in
 				if v, ok := storeMetrics["replicas.leaseholders"]; ok {
 					leases[i] += int(v.(float64))
 				}
-				if v, ok := storeMetrics["leasestransfers.success"]; ok {
+				if v, ok := storeMetrics["leases.transfers.success"]; ok {
 					leaseTransfers[i] += int(v.(float64))
 				}
 			}

--- a/pkg/cmd/internal/localcluster/localcluster.go
+++ b/pkg/cmd/internal/localcluster/localcluster.go
@@ -407,7 +407,7 @@ func (n *Node) Start() {
 		return
 	}
 
-	pid := n.cmd.ProcessState.Pid()
+	pid := n.cmd.Process.Pid
 	log.Infof(ctx, "process %d started: %s", pid, n.cmd.Args)
 
 	go func(cmd *exec.Cmd) {

--- a/pkg/util/humanizeutil/humanize.go
+++ b/pkg/util/humanizeutil/humanize.go
@@ -98,6 +98,13 @@ func (b *BytesValue) Type() string {
 
 // String implements the flag.Value and pflag.Value interfaces.
 func (b *BytesValue) String() string {
+	// We need to be able to print the zero value in order for go's flags
+	// package to not choke when comparing values to the zero value,
+	// as it does in isZeroValue as of go1.8:
+	// https://github.com/golang/go/blob/release-branch.go1.8/src/flag/flag.go#L384
+	if b.val == nil {
+		return IBytes(0)
+	}
 	// This uses the MiB, GiB, etc suffixes. If we use humanize.Bytes() we get
 	// the MB, GB, etc suffixes, but the conversion is done in multiples of 1000
 	// vs 1024.

